pkg/controller/pxc/controller_test.go

```diff
@@ -57,3 +69,450 @@ var _ = Describe("PerconaXtraDB Cluster", Ordered, func() {
 		Expect(err).To(Succeed())
 	})
 })
+
+var _ = Describe("Finalizer delete-ssl", Ordered, func() {
+	ctx := context.Background()
+
+	const crName = "del-ssl-fnlz"
+	const ns = "del-ssl-fnlz"
+	crNamespacedName := types.NamespacedName{Name: crName, Namespace: ns}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ns,
+			Namespace: ns,
+		},
+	}
+
+	BeforeAll(func() {
+		By("Creating the Namespace to perform the tests")
+		err := k8sClient.Create(ctx, namespace)
+		Expect(err).To(Not(HaveOccurred()))
+
+		_, err = envtest.InstallCRDs(cfg, envtest.CRDInstallOptions{
+			Paths: []string{filepath.Join("testdata", "cert-manager.yaml")},
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterAll(func() {
+		By("Deleting the Namespace to perform the tests")
+		_ = k8sClient.Delete(ctx, namespace)
+	})
+
+	Context("delete-ssl finalizer specified", Ordered, func() {
+
+		cr, err := readDefaultCR(crName, ns)
+		cr.Finalizers = append(cr.Finalizers, "delete-ssl")
+		cr.Spec.SSLSecretName = "cluster1-ssl"
+		cr.Spec.SSLInternalSecretName = "cluster1-ssl-internal"
+		It("should read default cr.yaml", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("Should create PerconaXtraDBCluster", func() {
+			Expect(k8sClient.Create(ctx, cr)).Should(Succeed())
+		})
+
+		When("PXC cluster is deleted with delete-ssl finalizer certs should be removed", func() {
+			It("should delete PXC cluster and reconcile changes", func() {
+				Expect(k8sClient.Delete(ctx, cr)).Should(Succeed())
+
+				_, err := reconciler().Reconcile(ctx, ctrl.Request{NamespacedName: crNamespacedName})
+				Expect(err).NotTo(HaveOccurred())
+			})
```

Comment:
Before this `When` container, we should reconcile and check if secrets, issuers and certificates were created

Reply:
0787ee2